contracts/bonding/BondingManager.sol

```diff
@@ -457,44 +457,48 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         whenSystemNotPaused
         onlyJobsManager
     {
-        // Transcoder must be valid
-        require(transcoderStatus(_transcoder) == TranscoderStatus.Registered);
-
         uint256 penalty = MathUtils.percOf(delegators[_transcoder].bondedAmount, _slashAmount);
         if (penalty > del.bondedAmount) {
             penalty = del.bondedAmount;
         }
 
         Delegator storage del = delegators[_transcoder];
 
-        // Decrease delegate's delegated amount
-        delegators[del.delegateAddress].delegatedAmount = delegators[del.delegateAddress].delegatedAmount.sub(penalty);
-        // Update total bonded tokens
-        totalBonded = totalBonded.sub(penalty);
-        // Decrease transcoder's stake
+        // Decrease bonded stake
         del.bondedAmount = del.bondedAmount.sub(penalty);
 
+        // If still bonded
+        // - Decrease delegate's delegated amount
+        // - Decrease total bonded tokens
+        if (delegatorStatus(_transcoder) == DelegatorStatus.Bonded) {
+            delegators[del.delegateAddress].delegatedAmount = delegators[del.delegateAddress].delegatedAmount.sub(penalty);
+            totalBonded = totalBonded.sub(penalty);
+        }
+
         uint256 currentRound = roundsManager().currentRound();
 
         if (activeTranscoderSet[currentRound].isActive[_transcoder]) {
-            // Set transcoder as inactive
-            activeTranscoderSet[currentRound].isActive[_transcoder] = false;
             // Decrease total active stake for the round
             activeTranscoderSet[currentRound].totalStake = activeTranscoderSet[currentRound].totalStake.sub(activeTranscoderTotalStake(_transcoder, currentRound));
+            // Set transcoder as inactive
+            activeTranscoderSet[currentRound].isActive[_transcoder] = false;
         }
 
-        // Remove transcoder from pools
-        transcoderPool.remove(_transcoder);
+        // If registered transcoder, remove from pool
+        if (transcoderStatus(_transcoder) == TranscoderStatus.Registered) {
+            transcoderPool.remove(_transcoder);
+        }
 
-        // Award finder fee
-        if (penalty > 0 && _finder != address(0)) {
```

Comment:
This was a bug - previously if there was not a finder address provided, `slashTranscoder` would not burn the slashed tokens. This change makes sure that `slashTranscoder` always burns the slashed tokens and if there is a finder address provided, it subtracts the finder's fee from the amount to be burned.